File: le-serviceflows/core/src/main/resources/com/latticeengines/serviceflows/core/transforms/std_visidb_ds_pd_alexa_relatedlinks_count.py

#
# $LastChangedBy$
# $LastChangedDate$
# $Rev$
#

def metadata():
    return {  'ApprovedUsage'   : 'Model'
            , 'DisplayDiscretizationStrategy' : '{"geometric": { "minValue":1,"multiplierList":[2,2.5,2],"minSamples":100,"minFreq":0.01,"maxBuckets":5,"maxPercentile":1}}'
            , 'Category'        : 'Online Presence'
            , 'FundamentalType' : 'numeric'
            , 'StatisticalType' : 'ratio'
            , 'Tags'            : 'External' }


def std_visidb_ds_pd_alexa_relatedlinks_count(n):
    if n is None or n == '':
        return None
    return n.count(',') + 1


def transform(args, record):
    column = args["column"]
    value = record[column]
    return std_visidb_ds_pd_alexa_relatedlinks_count(value)

File: le-transform/src/main/java/com/latticeengines/transform/v2_0_25/functions/StdVisidbDsPdAlexaRelatedlinksCount.java
package com.latticeengines.transform.v2_0_25.functions;

import java.util.Map;

import org.apache.commons.lang3.StringUtils;

import com.latticeengines.transform.exposed.RealTimeTransform;
import com.latticeengines.transform.exposed.metadata.ApprovedUsage;
import com.latticeengines.transform.exposed.metadata.Category;
import com.latticeengines.transform.exposed.metadata.FundamentalType;
import com.latticeengines.transform.exposed.metadata.StatisticalType;
import com.latticeengines.transform.exposed.metadata.Tag;
import com.latticeengines.transform.exposed.metadata.TransformMetadata;

public class StdVisidbDsPdAlexaRelatedlinksCount implements RealTimeTransform {

    private static final long serialVersionUID = -9101655382573322702L;

    public StdVisidbDsPdAlexaRelatedlinksCount() {
    }

    public StdVisidbDsPdAlexaRelatedlinksCount(String modelPath) {
    }

    public Object transform(Map<String, Object> arguments, Map<String, Object> record) {
        String column = (String) arguments.get("column");
        Object n = record.get(column);

        if (n == null)
            return null;

        String s = n.toString().trim().toLowerCase();

        return calculateStdVisidbDsPdAlexaRelatedlinksCount(s);
    }

    public static Integer calculateStdVisidbDsPdAlexaRelatedlinksCount(String alexaRelatedLinks) {
        if (StringUtils.isEmpty(alexaRelatedLinks))
            return null;

        return StringUtils.countMatches(alexaRelatedLinks, ",") + 1;
    }

    @Override
    public TransformMetadata getMetadata() {
        TransformMetadata metadata = new TransformMetadata();
        metadata.setApprovedUsage(ApprovedUsage.MODEL);
        metadata.setCategory(Category.ONLINE_PRESENCE);
        metadata.setFundamentalType(FundamentalType.NUMERIC);
        metadata.setStatisticalType(StatisticalType.RATIO);
        metadata.setTags(Tag.EXTERNAL_TRANSFORM);
        metadata.setDisplayDiscretizationStrategy("{\"geometric\": { \"minValue\":1,\"multiplierList\":[2,2.5,2],\"minSamples\":100," //
                + "\"minFreq\":0.01,\"maxBuckets\":5,\"maxPercentile\":1}}");
        return metadata;
    }
}
